src/reducers/ui.js

```diff
@@ -245,6 +281,11 @@ export const sGetUiRepetition = (state) => sGetUi(state).repetitionByDimension
 
 // Selectors level 2
 
+export const sGetUiInputType = (state) => sGetUiInput(state).type
+
+export const sGetUiProgramId = (state) => sGetUiProgram(state).id
+export const sGetUiProgramStage = (state) => sGetUiProgram(state).stage
```

Comment:
Related to my previous comment, if we went with storing the selected stage ID only, we could still have a selector called `sGetUiProgramStage` that gets the stage object, i.e. sth like this:
```js
export const sGetUiProgramStage = (state) =>
    state.ui.program.selectedStageId &&
    state.ui.program.programStages &&
    state.ui.program.programStages.find(
        ({ id }) => id === state.ui.program.selectedStageId
    )
```